### DISCUSSION TITLE
V11.0 Beta

### DISCUSSION THREAD

**Discussion by borgmanJeremy** [flameshot-org/flameshot#2191] [Q&A]
# V11.0 Beta

This is the BETA release for version 11. As you will see below there were major refactorings to the internals of Flameshot. We did our best internal testing, but this particular beta is more likely to have issues than most.

Please use the attached discussion to report any issues or discuss this beta. This will make it much easier for us to manage the beta issues. 

We will be in beta for 1-2 weeks depending on what kind of issues are found and then do the official release. 

## New Features
- The on-screen help menu has been clarified and dynamically updates the hotkeys based on user defined hotkeys. 

<p align=center><img src="https://github.com/flameshot-org/flameshot/blob/master/docs/images/help_screen.png" width=50%> </p>

- DBus is no longer required for CLI options.

- Flameshot can now be run in "one off" mode which means the background systray component is now optional. See the CLI details below.

- The CLI has been completely refactored. With the new architecture we added the following:
  - The man page has been rewritten to reflect all the new options. The best place to learn about these new features is the man page or flameshot --help, but some notable new features will be outlined here. 
  - CLI is now callable from MacOS.
  - CLI options are unified wherever possible. This means if an argument is added to "flameshot gui" it is also likely to be supported in "flameshot full".
  - `--region` is a new flag that allows users to specify the exact region to screenshot. It uses the same syntax as xrandr: `WxH+x+y`.
    - This is likely to be added to the launcher in the next release. 
  - `--accept-on-select` This flag will save the image as soon as the mouse is released when selecting a region.
  - The CLI now supports pinning (`--pin`), uploading (`--upload`), and copying to clipboard (`--clipboard`). 

- MacOS now uses monochrome icon to match the system theme better.

<p align=center><img src="https://github.com/flameshot-org/flameshot/blob/master/docs/images/monochrome.png" width=50%> </p>

- The sidebar now shows the hexidecimal color value when the color picker is used:
<p align=center><img src="https://github.com/flameshot-org/flameshot/blob/master/docs/images/hex.png" width=25%> </p>

- The about screen lists system information and allows you to copy this for easy access in bug submission forms. 

- Every file format supported by your underlying system is now an option for file format when saving. 

- New tool added to invert a region:

<p align=center><img src="https://github.com/flameshot-org/flameshot/blob/master/docs/images/invert.gif" width=75%> </p>

- Thickness of tools can now be set with the keyboard. Simply type a numerical value like "15" and you will see the indicator in the upper left.

- New zoom capability has been added to the color picker to more precisely select a color.
<p align=center><img src="https://github.com/flameshot-org/flameshot/blob/master/docs/images/magnify.gif" width=75%> </p>

- Text alignment can now be set in the side bar. 
<p align=center><img src="https://github.com/flameshot-org/flameshot/blob/master/docs/images/alignment.png" width=75%> </p>

- File names can now contain '.'

- Even if a button is hidden from the toolbar, it can still be activated via hotkey.

- The uploader now gives users a confirmation box before uploading. This can be disabled. **If you disable this and accidentally upload sensitive information, there is nothing we can do. It's recommended to leave the confirmation enabled.** Also, a keyboard shortcut for upload has been enabled (`Ctrl+U` by default)

- MacOS users can now bind a custom hotkey for taking a screenshot. 

- The config file parser has been reworked. It will now alert users if there is an error in their config. If a repair is possible, Flameshot attempts to repair the file. 
  - We do our best not to break existing configs, but sometimes adding new features or removing old ones force this to change. 

- Double clicking can be used to copy the screenshot to the clipboard. 

- Added an option to enable anti-aliasing when zooming in on a pinned image. 

## Bug Fixes 
- The border that indicates an object is moveable is no longer saved or copied with the underlying image. 

- The edit buttons no longer fall in the editable region when there are strange multi monitor geometries. 

- Optimizations to reduce lag on 8k and 4k screens.

- All Qt5 deprecations are fixed in preparation for Qt6.

- Many small UI improvements (ie oversized scrollbars fixed, checkboxes occluded, etc).

- Path handling has been improved.

- Fixed an issue where running Flameshot for the first time on NixOS would fail to create the config file.

- Fixed a problem with some window managers where Flameshot would lose focus and shortcuts would stop working.

## Known Issues
- Fractional scaling on linux is still not resolved.  (But we have identified a workaround finally. Hope to merge soon.)

## Contributors
I want to give special shout outs to some team members that made this release possible. 
- @veracioux for doing most of the refactoring that made this release possible 
- @mmahmoudian for tirelessly doing triage and community management
- @Correct-Syntax for the [redesigned website](https://flameshot.org)

We are very excited by the many first time contributors that helped with this release. We are always looking for more people to contribute to Flameshot and are happy to provide mentorship if needed:

* @johnjago made their first contribution in https://github.com/flameshot-org/flameshot/pull/1779
* @veracioux made their first contribution in https://github.com/flameshot-org/flameshot/pull/1782
* @etircopyh made their first contribution in https://github.com/flameshot-org/flameshot/pull/1799
* @uncomfyhalomacro made their first contribution in https://github.com/flameshot-org/flameshot/pull/1832
* @karlhorky made their first contribution in https://github.com/flameshot-org/flameshot/pull/1845
* @Cr4ckC4t made their first contribution in https://github.com/flameshot-org/flameshot/pull/1849
* @j-tai made their first contribution in https://github.com/flameshot-org/flameshot/pull/1856
* @CrystalSage made their first contribution in https://github.com/flameshot-org/flameshot/pull/1926
* @a1346054 made their first contribution in https://github.com/flameshot-org/flameshot/pull/1918
* @PrSunflower made their first contribution in https://github.com/flameshot-org/flameshot/pull/1582
* @mgalgs made their first contribution in https://github.com/flameshot-org/flameshot/pull/1940
* @GongHeng2017 made their first contribution in https://github.com/flameshot-org/flameshot/pull/1812
* @gVirtu made their first contribution in https://github.com/flameshot-org/flameshot/pull/1981
* @YizhePKU made their first contribution in https://github.com/flameshot-org/flameshot/pull/1979
* @Lyqst made their first contribution in https://github.com/flameshot-org/flameshot/pull/1995
* @AdavisSnakes made their first contribution in https://github.com/flameshot-org/flameshot/pull/1992
* @deo002 made their first contribution in https://github.com/flameshot-org/flameshot/pull/2008
* @Michael-F-Bryan made their first contribution in https://github.com/flameshot-org/flameshot/pull/2012
* @sryze made their first contribution in https://github.com/flameshot-org/flameshot/pull/2026
* @meesha7 made their first contribution in https://github.com/flameshot-org/flameshot/pull/2042
* @majkinetor made their first contribution in https://github.com/flameshot-org/flameshot/pull/2056
* @claytron5000 made their first contribution in https://github.com/flameshot-org/flameshot/pull/2068
* @LHBosssss made their first contribution in https://github.com/flameshot-org/flameshot/pull/2098
* @ffabss made their first contribution in https://github.com/flameshot-org/flameshot/pull/2140
* @reggermont made their first contribution in https://github.com/flameshot-org/flameshot/pull/2150
* @RiedleroD made their first contribution in https://github.com/flameshot-org/flameshot/pull/2130

<hr /><em>This discussion was created from the release <a href='https://github.com/flameshot-org/flameshot/releases/tag/v11.0.rc1'>V11.0 Beta</a>.</em>

**Comment by MrakoMaks** [comment]
Hi! There was a problem on one of the Windows 11 machines. The application starts (I see it in the task manager), but it is not in the tray. Reinstalling didn't help.

**Reply by borgmanJeremy** [reply]
Can you try deleting the config file (after making a backup)? If that solves it, post the config file so we can find the bug. If that doesnt fix it let me know. 

Did you install the portable version or the installer?

**Reply by majkinetor** [reply]
I had this problem before with 0.10.x version, but it disappeared without changing a version (Windows 10).  The only thing I remember is that I was tweaking startup apps.

**Reply by MrakoMaks** [reply]
@borgmanJeremy Thanks! Deleting the configuration file helped. When trying to import it, the following error occurs:
`flameshot: error: Semantic error in 'savePath'. Expected: existing directory`


**The configuration file:**
[General]
disabledTrayIcon=false
drawColor=#ff0000
drawThickness=3
showSidePanelButton=false
startupLaunch=true
contrastOpacity=188
savePath=C:/Users/\x41c\x430\x43a\x441\x438\x43c/Desktop
savePathFixed=false
showStartupLaunchMessage=false
buttons=@Variant(\0\0\0\x7f\0\0\0\vQList<int>\0\0\0\0\x12\0\0\0\0\0\0\0\x1\0\0\0\x2\0\0\0\x3\0\0\0\x4\0\0\0\x5\0\0\0\x6\0\0\0\x12\0\0\0\xf\0\0\0\x13\0\0\0\a\0\0\0\b\0\0\0\t\0\0\0\x10\0\0\0\n\0\0\0\v\0\0\0\f\0\0\0\x11)
showDesktopNotification=true
filenamePattern=flameshot_%F_%H-%M-%S

[Shortcuts]
TYPE_PENCIL=P
TYPE_DRAWER=D
TYPE_ARROW=A
TYPE_SELECTION=S
TYPE_RECTANGLE=R
TYPE_CIRCLE=C
TYPE_MARKER=M
TYPE_TEXT=T
TYPE_PIXELATE=B
TYPE_CIRCLECOUNT=
TYPE_SELECTIONINDICATOR=
TYPE_MOVESELECTION=Ctrl+M
TYPE_UNDO=Ctrl+Z
TYPE_REDO=Ctrl+Shift+Z
TYPE_COPY=Ctrl+C
TYPE_SAVE=Ctrl+S
TYPE_IMAGEUPLOADER=Return
TYPE_EXIT=Ctrl+Q
TYPE_PIN=
TYPE_TOGGLE_PANEL=Space
TYPE_RESIZE_LEFT=Shift+Left
TYPE_RESIZE_RIGHT=Shift+Right
TYPE_RESIZE_UP=Shift+Up
TYPE_RESIZE_DOWN=Shift+Down
TYPE_SELECT_ALL=Ctrl+A
TYPE_MOVE_LEFT=Left
TYPE_MOVE_RIGHT=Right
TYPE_MOVE_UP=Up
TYPE_MOVE_DOWN=Down
TYPE_COMMIT_CURRENT_TOOL=Ctrl+Return
TYPE_DELETE_CURRENT_TOOL=Del

**Reply by mmahmoudian** [reply]
@MrakoMaks it seems you have a lot of unicode characters as the username. If that correct? If not, try manually editing the config file and replace the `savePath` with the normal text of the username

**Reply by MrakoMaks** [reply]
> @MrakoMaks it seems you have a lot of unicode characters as the username. If that correct? If not, try manually editing the config file and replace the `savePath` with the normal text of the username

That was probably the problem. The settings were imported from another device at the time.
As I wrote above, fixed by deleting the config.

**Comment by veracioux** [comment]
Some things to fix:
- ~`flameshot launcher` (CLI) does not exit when the window is closed, tested on Linux~
- ~Same with upload in the `flameshot gui`~
- On MacOS, when zoom color grabber is active, keyboard shortcuts don't work (I'll work on this)

**Comment by mikiTesf** [comment]
I don't see the invert color option on my Ubuntu 20.04 machine? Is that feature limited to Mac OS an Windows at the moment?

**Reply by mmahmoudian** [reply]
No we always try to have features that are not platform-specific. The color invert is available to everyone. Have you checked that the button is enabled in the config? If not, right-click on the tray icon and select Configurations and scroll to find it. If you are on Gnome and don't have systray area, you can open configurations by running `flameshot config` in a terminal

**Reply by mikiTesf** [reply]
Thanks. I see it now. I enabled it in the configuration as you said.

**Comment by elad2109** [comment]
Thanks for the new release.

Two issues I've encountered on Debian:
1) After downloading the AppImage file, launching the flameshot doesn't add the systray.

`Flameshot can now be run in "one off" mode which means the background systray component is now optional. See the CLI details below."`

How do I refrain from "one off" mode?

2) I cannot save a capture as a pdf. Is it a bug or will it be possible in the future?

`Every file format supported by your underlying system is now an option for file format when saving.`

**Reply by borgmanJeremy** [reply]
2. Pdf is not an image format. I should have been more specific and said we support all image formats.


Edit:
1. By default it will be in the old mode. If the sts tray is not there please post all other info like desktop environment etc...appimage is the format I least test so something could be wrong there

**Reply by veracioux** [reply]
@elad2109
1. The flameshot background program (daemon) is what gives you the systray. Make sure the daemon is running by calling the AppImage without any arguments. Also make sure that the setting "Show tray icon" is enabled.
2. No, saving to PDF is not supported. The sentence you quoted was not very well phrased, sorry about that. You should read it as:
`Every IMAGE file format supported by your underlying system is now an option for file format when saving.`
Obviously we can't support every file format on the face of the earth :D.
Btw, I left a comment on the #2215 issue you posted.

**Reply by elad2109** [reply]
Thanks for the reply

1) I had set the custome keyboard shortcut to `./Flameshot-11.0.rc1.x86_64.AppImage gui` and after your comment I ran from the terminal `./Flameshot-11.0.rc1.x86_64.AppImage
^Z` as a bg process. The systray icon appears, none of the menu items work.

**Comment by SimonLafrikh** [comment]
Hi is there any info regarding the Gnome 41 + Wayland issue ? Is there something in progress to address the issue ?
https://github.com/flameshot-org/flameshot/issues/1910

**Reply by borgmanJeremy** [reply]
No, as you can see from that discussion this was an intentional change by the gnome project. You will need to work with them for any complaints. However, all the threads are locked because there was too much flaming/trolling/arguing going on. 

The real fix is to implement a true permissions portal in the xdg-portal. Everyone has agreed this is the right path forward but no one is actively working it.

**Reply by SimonLafrikh** [reply]
Ok thank you for confirming, I was referencing the xdg-portal implementation :)
